### ISSUE TITLE
Japanese POS Tag Support

### ISSUE THREAD

**Issue by polm** [explosion/spaCy#1214]
Would it be OK to add support for Japanese POS tags to spaCy? 

Janome, the library used for tokenization, already provides POS tags since they're generated as part of tokenization. That means this wouldn't require any extra dependencies or processing time over what's already being done when tokenizing Japanese.

If I wanted to implement this, is there anything in particular I should read first or keep in mind while working on it?

**Comment by honnibal** [comment]
This sounds very nice! A pull request would be very welcome.

The main thing to keep in mind is that you'll need to make sure that there are entries in the `Japanese.Defaults.tag_map` dictionary, to allow the `token.pos_` attribute to be computed. an example entry:
    
    from spacy.attrs import POS
    from spacy.symbols import NOUN

    `"NN": {POS: NOUN}

**Comment by polm** [comment]
Thanks, that sounds good.

I started working on a tag map but ran into an issue - turns out Japanese Universal Dependency tags aren't all direct mappings from raw tags, some require looking at the way a particular token is used in the sentence. This is documented in the notes at the bottom [here](http://universaldependencies.org/ja/overview/morphology.html#morphology).

These rules are also kind of unclear, and I haven't able to find any code that actually handles tag conversion, so I'm going to see if I can get some clarification from the Universal Dependencies folks.

**Comment by polm** [comment]
I'm still waiting on some information from the UD team on the mapping, but I've run into an issue - a simple dictionary like `tag_map` can't handle conversion from raw tags to Japanese UD tags. 

A paper introducing the Japanese UD project in English [here](http://www.ar.media.kyoto-u.ac.jp/publications/tanaka-LREC16.pdf) touches on the issue; to give an example:

![Same Unidic tag, different UD tag](https://user-images.githubusercontent.com/286278/28833025-20618cc8-7719-11e7-825a-21d341a3a057.png)

The "report" in the two sentences would have the same raw tag, but a different UD tag. (Another issue is resolving the UD tag would require a full dependency parse...)

Is there a way to deal with non-direct mappings like this in spaCy? Looking at existing tag maps I notice Spanish seems to append a lot of raw features into the source tag, but that won't necessarily resolve the ambiguity in this case.

**Comment by polm** [comment]
I'm closing this since the related changes were merged in #2036 and #1246. Thanks!
